podvm-mkosi/mkosi.skeleton/usr/lib/systemd/system/kata-agent.service.d/10-override.conf

```diff
@@ -0,0 +1,10 @@
+# On a read-only fs the kata-agent config is created in /run/peerpod, since it contains
+# a parameter that can be set at pod creation time.
+[Unit]
+ConditionKernelCommandLine=
+
+[Service]
+ExecStart=
+ExecStart=/usr/local/bin/kata-agent --config /run/peerpod/agent-config.toml
+ExecStop=
+ExecStopPost=/usr/local/bin/kata-agent-clean --config /run/peerpod/agent-config.toml
```

Comment:
The systemd file for process-user-data also needs to be overridden to specify the config file as `/run/peerpod/agent-config.toml` 

```
[Service]
ExecStart=
ExecStart=/usr/local/bin/process-user-data update-agent-config --agent-config-file /run/peerpod/agent-config.toml

```

Reply:
tested w/ mkosi. The unit doesn't fail anymore. (CDH still fails, but that's expected)